### PR TITLE
Updates to Python and C++ Docs

### LINKED ISSUE #2: Start doc for RAFT "principals"
Just a simple outline of principals we plan to follow in RAFT to help the ML and graph teams coordinate. 

- header only
- comms
- priorities for different versions (eg. what will be moved and when)
- abstraction paradigms (eg. resource management, etc...)

### PR DESCRIPTION
**Pull request by cjnolet**
As we're beginning to hide the implementation details and establisht he public API, it's time to start putting a little focus on the presentation of our docs. Eventually, we want to make sure our different packages are easy to navigate and find. We will be updating this more as we continue to stabilize the public API. 

The docs can be built by running `./build.sh docs` from the repository root. 

Closes #2

### CODE REVIEW

File: cpp/doxygen/main_page.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# libraft
+
+RAFT is an analytics engine, containing building-blocks for rapid composition of RAPIDS Analytics. These building-blocks include shared representations, mathematical computational primitives, and utilities that accelerate building analytics and data science algorithms in the RAPIDS ecosystem. Both the C++ and Python components can be included in consuming libraries, providing building-blocks for both dense and sparse matrix formats in the following general categories:
```
Comment (divyegala) [inline cpp/doxygen/main_page.md:3]: Can we maybe use a different term on the third "building-blocks"? Feels like a little repetitive term, we could use "representations"?

**Review by divyegala — APPROVED** [review]
Just one small comment, looks great otherwise

File: cpp/doxygen/main_page.md (hunk 1 above)
Comment (JohnZed) [reply cpp/doxygen/main_page.md:3]: Maybe not referring to it as an "engine" which sounds more like a server or executable to me? Library, framework, collection, building blocks, those are all great terms.

**Review by ajschmidt8 — COMMENTED** [review]
If these docs will eventually get published on our site, we'll need to add the `breathe` dependency to the `rapids-doc-env` below. @cjnolet, can you open a PR for that?

- https://github.com/rapidsai/integration/blob/3986715992ec3087f7e38e8b3dbf69035366874c/conda/recipes/rapids-doc-env/meta.yaml#L36

**Comment by cjnolet** [comment]
Absolutely, @ajschmidt8, I will do that.

**Review by ajschmidt8 — APPROVED** [review]
Approving `ops-codeowner` changes now that https://github.com/rapidsai/integration/pull/415 is merged.

**Review by BradReesWork — APPROVED** [review]

**Comment by cjnolet** [comment]
@gpucibot merge

**Pull request merged**